src/components/Observations/map/ObservationMapView.tsx

```diff
@@ -122,6 +134,32 @@ export default function ObservationMapView({ observationsResults, search }: Obse
     ]
   );
 
+  const contextRenderer = (properties: MapSourceProperties): JSX.Element => {
+    let entity: any;
+    if (properties.type === 'site') {
+      entity = selectedObservation;
+    } else if (properties.type === 'zone') {
+      entity = selectedObservation?.plantingZones?.find((z) => z.plantingZoneId === properties.id);
+    } else {
+      // monitoring plot
+      entity = selectedObservation?.plantingZones
+        ?.flatMap((z) => z.plantingSubzones)
+        ?.flatMap((sz) => sz.monitoringPlots)
+        ?.find((p) => p.monitoringPlotId === properties.id);
+    }
+
```

Comment:
Guessing we will always have an entity here.. would we ever need to return null here in case there's no entity? guessing mapbox won't call this code if a source/geometry was not clicked.

Reply:
The values in properties (id, name, etc) _should_ always correspond to an actual entity.

Reply:
Got it, makes sense.. the callback only happens on clicked objects.